lib/events/filesessions/filestream.go

```diff
@@ -384,6 +395,46 @@ func checkUploadID(uploadID string) error {
 	return nil
 }
 
+// generateReservationFileContent generates the content placed on the
+// reservation files.
+func generateReservationFileContents(partNumber int64) ([]byte, error) {
+	// Create a buffer with the max size that a part file can have.
+	buf := make([]byte, minUploadBytes+events.MaxProtoMessageSizeBytes)
+
+	// Encode reservation content.
+	encoded := &bytes.Buffer{}
+	encoder := gob.NewEncoder(encoded)
```

Comment:
What is going on here? Why do we need to introduce yet another encoding format?

Reply:
I wasn't sure what encoding to use to identify those "reservation" files. First, I've added a Protobuf message, but it seemed like a little of an overkill for this since we're not using the data inside those messages. Do you think we should still use Protobuf for this, or do you have any other suggestions? The idea is to have some content on those files to differentiate them from regular session recording parts (Protobuf encoded).

Reply:
Why not just rename them to `<uuid>.reserve` or something along those lines?

Reply:
> Why not just rename them to <uuid>.reserve or something along those lines?

I think that it is a good alternative because the [`ListParts`](https://github.com/gravitational/teleport/blob/3ffe935f478e51a339257464e5de0f0e28e398b6/lib/events/filesessions/filestream.go#L353) wouldn't recognize those files (not causing it to create broken session recordings) and they will be removed by the [`CompleteUpload`](https://github.com/gravitational/teleport/blob/master/lib/events/filesessions/filestream.go#L184).